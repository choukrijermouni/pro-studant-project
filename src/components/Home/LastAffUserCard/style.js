import { blue } from '@pro_boa/ui'

export default {
  cardContainer: {
    cursor: 'pointer',
    transition: 'transform 0.2s ease',
    '&:hover': {
      transform: 'scale(1.01) translateY(-1px)'
    },
    '&:hover $avatarContainer': {
      transform: 'scale(1.15)'
    }
  },
  row: {
    marginBottom: 8,
    display: 'flex',
    justifyContent: 'space-between',
    minHeight: 50,
    overflow: 'hidden',
    alignItems: 'center',
    padding: 10
  },
  nameContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    cursor: 'pointer'
  },
  nameBox: {
    width: '11vw'
  },
  card: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    marginLeft: 24
  },
  icon: {
    fontSize: 16,
    color: blue[1]
  },
  avatarContainer: {
    transition: 'transform 0.2s ease'
  }
}
