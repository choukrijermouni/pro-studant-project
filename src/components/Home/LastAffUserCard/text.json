{
  "neverConnected": "Aucune activité",
  "invited": "Invité",
  "lastAffectation": "Date d’affectation"
}