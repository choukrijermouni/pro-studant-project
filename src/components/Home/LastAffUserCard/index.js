import React from 'react'
import { createUseStyles } from 'react-jss'
import style from './style'
import { Paper, AvatarSize, Avatar, Icon, Icons, TypographyElement, neutral } from '@pro_boa/ui'
import moment from 'moment'
import { DMYDateFormat, amazonBucket } from 'constants/'
import { useDispatch, useSelector } from 'react-redux'
import { push } from 'connected-react-router'
import { LearnerProfileNeutralPath } from 'Routes'
import { avatarName } from 'helpers'
import Text from './text.json'

export default ({ data, license }) => {
  const dispatch = useDispatch()
  const useStyle = createUseStyles(style)
  const fullName = `${data?.FirstName} ${data?.LastName}`
  const { row, nameContainer, icon, card, nameBox, cardContainer, avatarContainer } = useStyle()
  const { organizationLicenseTypes } = useSelector(state => state.referential)
  return (
    <div
      data-test='last-assign-user-card'
      className={cardContainer}
      onClick={() => dispatch(push(`${LearnerProfileNeutralPath}/${data?.UserId}`))}
    >
      <Paper className={row}>
        <div className={nameContainer}>
          <div className={avatarContainer}>
            <Avatar
              name={avatarName(fullName)}
              img={data.Photo ? `${amazonBucket.bucketBaseUrl}${data.Photo}` : null}
              size={AvatarSize.size_1}
            />
          </div>
          <TypographyElement
            component='h3'
            variant='heading3'
            align='left'
            fontSize={13}
            spacing='20px'
            color={neutral[6]}
            className={nameBox}
          >
            {`${data?.FirstName || ''} ${data?.LastName || ''}`}
          </TypographyElement>
        </div>
        <div title={Text.lastAffectation} className={card}>
          <Icon iconName={Icons.lastUserLogin} style={icon} />
          <TypographyElement
            component='h4'
            variant='caption1'
            align='left'
            spacing='10px 16px 11px 16px'
            color={neutral[4]}
          >
            {moment(data?.OperationDate).format(DMYDateFormat)}
          </TypographyElement>
        </div>
        <Avatar
          name={organizationLicenseTypes[data?.OrganizationLicenseType]?.charAt(0)?.toUpperCase()}
          size={AvatarSize.size_1}
        />
      </Paper>
    </div>
  )
}
