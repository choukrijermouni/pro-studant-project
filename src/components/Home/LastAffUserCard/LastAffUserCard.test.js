import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import LastAffUserCard from '.'
configure({ adapter: new Adapter() })

describe('>>> LastAffUserCard --- Shallow Render ManagerCard component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<LastAffUserCard />)
    tree = renderer.create(
      <LastAffUserCard />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
