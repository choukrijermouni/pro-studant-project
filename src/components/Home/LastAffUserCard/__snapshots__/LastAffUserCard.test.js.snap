// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> LastAffUserCard --- Shallow Render ManagerCard component +++ render correctly and match snapshot 1`] = `
<div
  className="row-0-2-5 paper-0-2-9"
>
  <div
    className="nameContainer-0-2-6"
  >
    <div
      className="avatarContainer-0-2-11"
    >
      <div
        className="dot-0-2-10"
      />
      <span
        className="avatarGroup-0-2-12 avatarGroup-d0-0-2-13"
      >
        YS
      </span>
    </div>
    <p
      className="text-0-2-14 text-d0-0-2-15"
    >
      choukri jermouni
    </p>
  </div>
  <div
    className="card-0-2-7"
  >
    <span
      className="icon-last-login-user-alph-iconfont icon-0-2-8"
    />
    <p
      className="text-0-2-16 text-d0-0-2-17"
    >
      01/01/2020
    </p>
  </div>
</div>
`;
